test/parser/parser_spec.coffee

```diff
@@ -0,0 +1,12 @@
+chai = require 'chai'
+expect = chai.expect
+
+parse = require '../../parser/parser'
+outputTokens = require '../example/output_tokens'
+
+describe 'Scanner', ->
+
+  describe 'parsing valid teascript programs', ->
+    describe 'valid teascript program #1', ->
+      context "when test program1.tea's tokens
+               are passed through the scanner", ->
```

Comment:
It is possible to do bottom-up TDD where each of the parsing functions have a test that takes in a (mocked) token array.